viewer/packages/camera-manager/src/Flexible/FlexibleCameraManager.ts

```diff
@@ -0,0 +1,551 @@
+    this._isEnabled = value;
+    this.controls.isEnabled = true;
+  }
+
+  private getPosition(): Vector3 {
+    return this.camera.position.clone();
+  }
+
+  private getTarget(): Vector3 {
+    return this.controls.getTarget();
+  }
+
+  public setPositionAndTarget(position: Vector3, target: Vector3): void {
+    if (this.controls.isEnabled) {
+      this.controls.setState(position, target);
+    }
+  }
+
+  //================================================
+  // INSTANCE METHODS: Calculations
+  //================================================
+
+  private async getPickedPointPixelCoordinates(pixelX: number, pixelY: number): Promise<Vector3> {
+    const raycastResult = await this._raycastCallback(pixelX, pixelY, false);
+    if (raycastResult.intersection?.point) {
+      return raycastResult.intersection.point;
+    }
+    return this.getTargetByBoundingBox(pixelX, pixelY, raycastResult.modelsBoundingBox);
+  }
+
+  private getTargetByBoundingBox(pixelX: number, pixelY: number, boundingBox: Box3): Vector3 {
+    const raycaster = new Raycaster();
+    const pixelCoordinates = getNormalizedPixelCoordinates(this.domElement, pixelX, pixelY);
+    raycaster.setFromCamera(pixelCoordinates, this.camera);
+
+    // Try to intersect the bounding box
+    const closestIntersection = new Vector3();
+    const furthestIntersection = new Vector3();
+    const intersectionCount = intersectBox(raycaster.ray, boundingBox, closestIntersection, furthestIntersection);
+    if (intersectionCount === 1) {
+      return closestIntersection;
+    }
+    if (intersectionCount === 2) {
+      return furthestIntersection;
+    }
+    // Fallback to the old method
+    const diagonal = getDiagonal(boundingBox);
+    const center = boundingBox.getCenter(new Vector3());
+    const distanceToCenter = this.camera.position.distanceTo(center);
+    const distance = Math.min(distanceToCenter, diagonal) / 2;
+    return raycaster.ray.at(distance, new Vector3());
+  }
+
+  //================================================
+  // INSTANCE METHODS: Set up/tear down events
+  //================================================
+
+  private addEventListeners() {
+    this._inputHandler.on('click', this.onClick);
+    this.domElement.addEventListener('keydown', this.onKeyDown);
+    this.domElement.addEventListener('dblclick', this.onDoubleClick);
+    this.domElement.addEventListener('wheel', this.onWheel);
+  }
+
+  private removeEventListeners(): void {
+    this._inputHandler.off('click', this.onClick);
+    this.domElement.removeEventListener('dblclick', this.onDoubleClick);
+    this.domElement.removeEventListener('keydown', this.onKeyDown);
+    this.domElement.removeEventListener('wheel', this.onWheel);
+  }
+
+  //================================================
+  // INSTANCE METHODS: Event Handlers
+  //================================================
+
+  private readonly onKeyDown = (event: KeyboardEvent) => {
+    if (!this.isEnabled) {
```

Comment:
Not critical, but I'm starting to wonder if there's a cleaner way to handle that the controller is disabled than by doing this check so many places :thinking: